CMakeLists.txt

```diff
@@ -123,6 +133,70 @@ if(USE_GPU)
     ADD_DEFINITIONS(-DUSE_GPU)
 endif(USE_GPU)
 
+if(USE_CUDA)
+    find_package(CUDA REQUIRED)
+    include_directories(${CUDA_INCLUDE_DIRS})
+    LIST(APPEND CMAKE_CUDA_FLAGS -g -Xcompiler=-fopenmp -Xcompiler=-fPIC -Xcompiler=-Wall -lineinfo)
+    CUDA_SELECT_NVCC_ARCH_FLAGS(CUDA_ARCH_FLAGS 7.0)
```

Comment:
Could you please explane why is `7.0` used here? Do we need something similar to https://github.com/dmlc/xgboost/blob/d39da42e699c820f07778b690379cb529e0ad885/cmake/Utils.cmake#L86-L111 or any other more modern solution?

Reply:
It seems the cmake documentation is a little out of date. 7.0 is an acceptable value for the required API in CUDA files.

Reply:
Oh, sorry, seems that I was not clear enough in my previous comment! I wanted to know why `7.0` but not, let say, `6.5` or any other number.

Reply:
I'm adding all architectures from 6.0 onward.  6.0 is needed because of the way atomics are handled.